Index: gcc/libstdc++-v3/crossconfig.m4
===================================================================
--- gcc.orig/libstdc++-v3/crossconfig.m4
+++ gcc/libstdc++-v3/crossconfig.m4
@@ -299,6 +299,17 @@ dnl # functions in the list guarded by
 dnl # long_double_math_on_this_cpu in configure.ac, right after
 dnl # the expansion of the present macro.
     ;;
+  *-mlibc*)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_LINKER_FEATURES
+    GLIBCXX_CHECK_MATH_SUPPORT
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    AC_DEFINE([_GLIBCXX_USE_DEV_RANDOM])
+    AC_DEFINE([_GLIBCXX_USE_RANDOM_TR1])
+    GCC_CHECK_TLS
+    AC_CHECK_FUNCS([aligned_alloc posix_memalign memalign _aligned_malloc])
+    AC_CHECK_FUNCS([timespec_get])
+    ;;
   *)
     AC_MSG_ERROR([No support for this host/target combination.])
    ;;
